### DISCUSSION TITLE
[REQUEST] Add hibernation mode

### DISCUSSION THREAD

**Discussion by SilentMystification** [eduardofilo/RG350_adam_image#292] [General]
OnionOS on the Miyoo Mini has a wonderful feature where after 5-10 minutes of being in sleep mode it would automatically enter hibernation mode. When powered back on, you would be plopped back where you were when you entered sleep mode.

This is such a fantastic feature there as I've noticed that  battery life in sleep mode isn't the best and I accidentally turn the console on in my pocket all the time when carrying it atround.

Would this be possible to add as an option or as a hibernation script?

**Comment by eduardofilo** [comment]
It would be nice of course, but Adam is not a development project, but a combined distribution of system, frontend and emulators. Also, I don't have the knowledge to know how to detect a period of inactivity to try to trigger ODbeta's sleep mode.
Sorry. I wish I could give you another answer.

**Reply by SilentMystification** [reply]
Ahh that's too bad! Regardless, thanks for the response :)

**Comment by curist** [comment]
Inspired by https://www.reddit.com/r/RG350/comments/oe9jyf/deep_sleep_mode_for_rg350_rg280_retroarch_only/
and the ergonomics of Miyoo Mini, I put up something like below to achieve similar behavior for rg280.

Please keep in mind all path below are rg280v specific.

**/sbin/poweroff**
```sh
#!/bin/sh

PS=$(ps | grep retroarch | awk 'NR==2')

AUTOBOOT=/media/data/local/home/autoboot.sh

if [ -z "$PS" ]; then
  rm $AUTOBOOT
else
  killall -15 retroarch_rg350_odbeta
  HISTORY_LIST="$HOME/.retroarch/content_history.lpl"
  ROMINFO="$HOME/rominfo-filter.txt"
  awk -F": " 'NR==11{print $2, $3}' $HISTORY_LIST > $ROMINFO
  awk -F": " 'NR==13{print $2, $3}' $HISTORY_LIST >> $ROMINFO

  PATH_GAME=$(awk -F"\"" 'NR==1{print $2}' $ROMINFO)
  CORE_PATH=$(awk -F"\"" 'NR==2{print $2}' $ROMINFO)

  rm $ROMINFO

  echo "#!/bin/sh" > $AUTOBOOT
  echo "/usr/local/bin/retroarch_rg350_odbeta -L \"$CORE_PATH\" \"$PATH_GAME\"" >> $AUTOBOOT
  chmod +x $AUTOBOOT
fi

/usr/bin/busybox poweroff
```

**/etc/pwswd.conf**
```
# /etc/pwswd.conf
# Default configuration for pwswd,
# OpenDingux' Power Switch Daemon.
#
# Values can be overwritten in /usr/local/etc/pwswd.conf

[Shortcuts]

#REBOOT			= START,SELECT
#SUSPEND			= X
VOLUME_UP		= UP
VOLUME_DOWN		= DOWN
BRIGHTNESS_UP	= RIGHT
BRIGHTNESS_DOWN	= LEFT
MOUSE_EMULATION	= B
#TV_OUT			= X # Not working yet
SCREENSHOT		= Y
KILL			= SELECT

RATIOMODE		= A

# The "poweroff" backend is now automatically called when the power slider
# is held for more than 3 seconds. To disable that feature and use a key
# combo instead, set the POWEROFF variable.
POWEROFF		= START
```

**/media/data/local/home/.autostart**
```sh
#!/bin/sh

# author MashTec
# modified by FGL82

if [ -f /media/data/local/home/autoboot.sh ]; then
    /media/data/local/home/autoboot.sh
    rm /media/data/local/home/autoboot.sh
fi

if [ -f /media/data/apps/SimpleMenu-OD-BETA.opk ]; then
    /usr/bin/opkrun -m default.gcw0.desktop /media/data/apps/SimpleMenu-OD-BETA.opk
elif [ -f /media/sdcard/apps/SimpleMenu-OD-BETA.opk ]; then
    /usr/bin/opkrun -m default.gcw0.desktop /media/sdcard/apps/SimpleMenu-OD-BETA.opk
elif [ -f /media/sdcard/APPS/SimpleMenu-OD-BETA.opk ]; then
    /usr/bin/opkrun -m default.gcw0.desktop /media/sdcard/APPS/SimpleMenu-OD-BETA.opk
else
    /usr/bin/gmenu2x
fi
```

After having these all in place, we can now press `power + start` in game to poweroff, and will later boot directly into last played retroarch rom.

@eduardofilo not sure if you would like to integrate these into future release?

EDIT: Got to update `scripts/frontend_start` of [SimpleMenu-OD-BETA.opk](https://github.com/eduardofilo/RG350_adam_image/blob/master/data/apps/SimpleMenu-OD-BETA.opk) to let SimpleMenu correctly to identify if itself is default launcher.

**Reply by eduardofilo** [reply]
WOW. Very interesting. I will definitely look into including it in the next release.
Thank you very much.
